### PR TITLE
cov scripts and small changes in signed

### PR DESCRIPTION
**Pull request by mike1729**
- scripts for code coverage generation
- more tests in signed
- some fixes in signed

### CODE REVIEW

File: src/signed.rs [hunk 1]
```diff
@@ -206,15 +201,15 @@ pub struct IncompleteMultisignatureError<'a, T: Signable, MK: MultiKeychain> {
 }
 
 impl<'a, T: Signable, MK: MultiKeychain> PartiallyMultisigned<'a, T, MK> {
-    pub fn sign(signable: T, keychain: &'a MK) -> Self {
+    pub fn sign(signable: Indexed<T>, keychain: &'a MK) -> Self {
```
Comment (timorl) [inline src/signed.rs:204]: This shouldn't require an `Indexed` argument, anything can be signed and the wrapper is not even used for anything.

Comment (timorl) [reply src/signed.rs:204]: You can, you should just pass the underlying type here, not the wrapped type you had to pass to create the `Signed` object.

I actually had a discussion about this with Michał and we ended up creating a low priority task to make explicit usage of `Indexed` completely unnecessary.

Comment (michalseweryn) [reply src/signed.rs:204]: `add_signature` takes `&mut PartiallyMultisigned<..., T, ...>` and `Signed<..., Indexed<T>, ...>`. Your version is a bit redundant, and you probably should anyway assert that `signable.index` is equal to `keychain.index()`

Comment (mike1729) [reply src/signed.rs]: Meh...you're absolutely right. I totally misunderstood some compiler error and followed it wrongly. I've reverted the change.

File: src/signed/tests.rs [hunk 2]
```diff
@@ -117,28 +137,68 @@ impl MultiKeychain for TestMultiKeychain {
...
+fn test_incomplete_multisignature() {
+    let msg = indexed_test_message(0);
+    let index: NodeIndex = 0.into();
+    let node_count: NodeCount = 2.into();
+    let keychain = TestMultiKeychain { node_count, index };
+
+    let partial = PartiallyMultisigned::sign(msg, &keychain);
+    assert!(
+        partial._try_into_complete(&keychain).is_err(),
+        "5 signatures should form a complete signature"
```
Comment (michalseweryn) [inline src/signed/tests.rs:182]: Probably a message "1 signature does not form a complete signature" is more descriptive?

Comment (mike1729) [reply src/signed/tests.rs:182]: Definitely.

**Pull request merged**